client/src/components/ProfileEdit.jsx

```diff
@@ -7,94 +7,90 @@ import { Link } from 'react-router-dom'
+                "location": this.state.location_input,
+                "password": this.state.new_password_input,
+                "id": this.state.user.id
             }
-            request('/profile/' + this.props.match.params.profile_id, "PUT", JSON.stringify(data))
+            request('/profile/' + this.props.match.params.profile_id,
+                    "PUT", JSON.stringify(data))
             .then(data => {
-                try {
-                    if ('errors' in data) {
-                        for (let i = 0; i < data['errors'].length; i++) {
-                            if (data.errors[i].name === "first_name") {
-                                this.setState({
-                                    error_first: data.errors[i].description
-                                })
-                            }
-                            if (data.errors[i].name === "last_name") {
-                                this.setState({
-                                    error_last: data.errors[i].description
-                                })
+                if ('errors' in data) {
+                    for (let i = 0; i < data['errors'].length; i++) {
+                        this.setState({
+                            errors_edit: {
+                                ...this.state.errors_edit,
+                                [data.errors[i].name]:
+                                String([data.errors[i].description])
                             }
-                            if (data.errors[i].name === "nickname") {
-                                this.setState({
-                                    error_nick: data.errors[i].description
-                                })
-                            }
-                        };
-                    }
-                    else {
-                        this.setState({ error_nick: "" })
-                        this.setState({ error_repeated: ""})
-                        this.setState({ error_first: ""})
-                        this.setState({ error_last: ""})
-                        this.props.history.push('/profile/' + this.props.match.params.profile_id);
-                        window.Materialize.toast("Profile Updated", 2500);
-                    }
+                        });
+                    };
+                }
+                else if (!data.success) {
+                    this.setState({
+                        errors_edit: {}
+                    });
+                    this.props.history.push('/profile/' + this.props.match.params.profile_id);
+                    window.Materialize.toast("Profile Updated", 2500);
                 }
-                catch (err) {
-                    this.setState({ error_nick: "" })
-                    this.setState({ error_repeated: ""})
-                    this.setState({ error_first: ""})
-                    this.setState({ error_last: ""})
+                else {
+                    sessionStorage.setItem("User-nickname",
```

Comment:
This is one place where you must rewrite nickname in session storage, remove another